lib/text/ui_text_displayer.js

```diff
@@ -705,7 +747,7 @@ shaka.text.UITextDisplayer = class {
 
     style.webkitTextStrokeColor = cue.textStrokeColor;
     style.webkitTextStrokeWidth = cue.textStrokeWidth;
-    style.color = cue.color;
+    style.color = this.readabilityTextColor_(cue.color);
```

Comment:
Suggestion to drop a separate function for each property. You can add a default styles map to [player_configuration.js](https://github.com/shaka-project/shaka-player/blob/main/lib/util/player_configuration.js#L447-L450). This would make it a one liner I believe: `style.color = cue.color || this.config_.textColor`